guides/common/modules/proc_installing-puppet-agent-during-host-provisioning.adoc

```diff
@@ -1,30 +1,56 @@
 [id="Installing_Puppet_Agent_during_Host_Provisioning_{context}"]
 = Installing Puppet Agent during Host Provisioning
 
-Install the Puppet agent during the host provisioning process.
+Use this procedure to install the Puppet agent during the host provisioning process.
+
+ifdef::orcharhino,satellite[]
+.Prerequisites
+* You enabled and synchronized the *{project-client-name}* repository to {Project}.
+For more information, see {ContentManagementDocURL}Importing_Content_content-management[Importing Content] in _{ContentManagementDocTitle}_.
+* You created an activation key that enables the *{project-client-name}* repository for hosts.
+For more information, see {ContentManagementDocURL}Managing_Activation_Keys_content-management[Managing Activation Keys] in _{ContentManagementDocTitle}_.
+* You assigned the activation key to a host group.
+endif::[]
+ifdef::katello[]
+.Prerequisites
+* You created a Product and repository for the upstream Puppet platform, such as `\https://yum.puppet.com` or `\https://apt.puppet.com`, and synchronized the repository to {Project}.
+For more information, see {ContentManagementDocURL}Importing_Content_content-management[Importing Content] in _{ContentManagementDocTitle}_.
+* You created an activation key that enables the repository for hosts.
+For more information, see {ContentManagementDocURL}Managing_Activation_Keys_content-management[Managing Activation Keys] in _{ContentManagementDocTitle}_.
+* You assigned the activation key to a host group.
+endif::[]
 
 .Procedure
 . Navigate to *Hosts > Provisioning Templates*.
 . Select a provisioning template depending on your host provisioning method.
 For more information, see {ProvisioningDocURL}types-of-provisioning-templates_provisioning[Types of Provisioning Templates] in _{ProvisioningDocTitle}_.
+ifndef::katello,orcharhino,satellite[]
+. Ensure the `puppetlabs_repo` snippet is included as follows:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+<%= snippet 'puppetlabs_repo' %>
+----
+endif::[]
 . Ensure the `puppet_setup` snippet is included as follows:
 +
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
 <%= snippet 'puppet_setup' %>
 ----
-. Enable Puppet using a host parameter for a single host or host group:
+ifndef::katello,orcharhino,satellite[]
+. Enable the official Puppet repository using a host parameter in global parameters, a host group, or for a single host.
 +
-Add a host parameter named `enable-puppet7` for Puppet 7
-ifndef::satellite[]
-or `enable-puppet6` for Puppet 6
+Add a host parameter named `enable-puppetlabs-repo` for the latest Puppet 7 or `enable-puppetlabs-puppet6-repo` for Puppet 6, select the `boolean` type, and set the value to `true`.
```

Comment:
We have 4 versions now:
* `enable-puppetlabs-repo`: unversioned repository, latest stable release (so will be updated to Puppet 8 when it comes out later this year)
* `enable-official-puppet7-repo`: Puppet 7, latest current stable now
* `enable-puppetlabs-puppet6-repo`: Puppet 6, going EOL soon
* `enable-puppetlabs-puppet5-repo`: Puppet 5, already EOL

Probably shouldn't document the EOL ones, but giving the full overview.